--- kdebindings-3.3.0/dcopperl/Makefile.PL~	2004-08-28 17:23:49.247677128 +0200
+++ kdebindings-3.3.0/dcopperl/Makefile.PL	2004-08-28 17:24:16.094595776 +0200
@@ -17,6 +17,7 @@
     XS		=> {'DCOP.xs' => 'DCOP.cpp'},
     XSOPT	=> '-C++',
     CCFLAGS	=> '-x c++',
+    INSTALLDIRS => 'vendor',
 );
 
 sub MY::xs_c {
--- kdebindings-3.3.0/dcopperl/Makefile.PL.in~	2004-08-28 17:35:27.162577936 +0200
+++ kdebindings-3.3.0/dcopperl/Makefile.PL.in	2004-08-28 17:34:57.554079112 +0200
@@ -17,6 +17,7 @@
     XS		=> {'DCOP.xs' => 'DCOP.cpp'},
     XSOPT	=> '-C++',
     CCFLAGS	=> '-x c++',
+    INSTALLDIRS => 'vendor',
 );
 
 sub MY::xs_c {
--- kdebindings-3.3.0/admin/acinclude.m4.in.orig	2004-08-30 14:00:55.242003048 +0200
+++ kdebindings-3.3.0/admin/acinclude.m4.in	2004-08-30 14:01:34.344058632 +0200
@@ -4170,13 +4170,7 @@
   LIBPYTHON=-lpython$version
 fi
 
-AC_FIND_FILE(python$version/copy.py, $python_libdirs, python_moddir)
-python_moddir=$python_moddir/python$version
-if test ! -r $python_moddir/copy.py; then
-  python_moddir=no
-fi
-
-PYTHONMODDIR=$python_moddir
+PYTHONMODDIR=$ac_python_dir
 
 AC_MSG_RESULT(header $python_incdir library $python_libdir modules $python_moddir)
 
