--- kdebindings-3.3.0/admin/acinclude.m4.in.orig	2005-01-10 22:08:22.121071912 +0100
+++ kdebindings-3.3.0/admin/acinclude.m4.in	2005-01-10 22:08:34.582177536 +0100
@@ -4151,7 +4151,7 @@
 
 PYTHONINC=-I$python_incdir
 
-python_libdirs="$ac_python_dir/lib$kdelibsuff /usr/lib$kdelibsuff /usr/local /usr/lib$kdelibsuff $kde_extra_libs"
+python_libdirs="$ac_python_dir/lib64$kdelibsuff /usr/lib64$kdelibsuff /usr/local /usr/lib64$kdelibsuff $kde_extra_libs"
 AC_FIND_FILE(libpython$version.so, $python_libdirs, python_libdir)
 if test ! -r $python_libdir/libpython$version.so; then
   AC_FIND_FILE(libpython$version.a, $python_libdirs, python_libdir)
